src/data-connect/data-connect-api.ts

```diff
@@ -0,0 +1,56 @@
+/*!
+ * @license
+ * Copyright 2024 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Interface representing a Data Connect connector configuration.
+ */
+export interface ConnectorConfig {
+  /**
+   * Location ID of the Data Connect service.
+   */
+  location: string;
+
+  /**
+   * Service ID of the Data Connect service.
+   */
+  serviceId: string;
+}
+
+/**
+ * Interface representing GraphQL response.
+ */
+export interface ExecuteGraphqlResponse<GraphqlResponse> {
+  /**
+   * Data payload of the GraphQL response.
+   */
+  data: GraphqlResponse;
+}
+
+/**
+ * Interface representing GraphQL options.
+ */
+export interface GraphqlOptions<Variables> {
+  /**
+   * Values for GraphQL variables provided in this query or mutation.
+   */
+  variables?: Variables;
```

Comment:
I wonder if we should make this required, and then have an additional signature that has `GraphqlOptions<undefined>` and accepts no arguments?

Reply:
Like: https://github.com/firebase/firebase-js-sdk/blob/main/packages/data-connect/src/api/query.ts#L91-L120
Curious to hare @hsubox76's opinion

Reply:
I am not sure I understood correctly... `GraphqlOptions` here is an interface and the link to js-sdk points to function signatures. What would be the use-case here for marking `variables` in `GraphqlOptions` required?

Reply:
Say a user defines `GraphqlOptions<number>`, based on the `executeGraphql` function, if the user uses that type, then they can optionally not provide a number. Do we want that? I feel like if a developer defines a variables type, then they want to enforce that that type is being passed in, and a way to do that is to make the Variables required in the type itself, and if developers don't want to pass something in, they can also completely leave out the Variables object.
Here's an example:
Currently:
```ts
executeGraphqlRead<number>(myQuery); // this is valid
```

We could change the signature so that the user has to pass in a variable if they pass in a type. WDYT?